code/modules/surgery/organs/internal/appendix/_appendix.dm

```diff
@@ -72,19 +107,74 @@
+	var/mob/living/carbon/organ_owner = owner
+	if(inflamation_stage < EXPLOSIVE_MAX_STAGE && SPT_PROB(INFLAMATION_ADVANCEMENT_PROB * 0.5, seconds_per_tick))
+		inflamation_stage += 1
+
+	if(get_stabilized_status() && inflamation_stage < EXPLOSIVE_TERMINAL_STAGE)
+		inflamation_stage = 0
+		appendicitis_type = null
+		to_chat(organ_owner, span_notice("Your chest feels like it's completely stabilized. It doesn't hurt at all!"))
+		return
+
+	switch(inflamation_stage)
+		if(1 to 2)
+			if(SPT_PROB(0.5, seconds_per_tick))
+				to_chat(organ_owner, span_warning("Your chest hurts miserably, as if something was pressing on it very heavily."))
+				INVOKE_ASYNC(organ_owner, TYPE_PROC_REF(/mob, emote), "grimace")
+		if(EXPLOSIVE_TERMINAL_STAGE) // point of no return
+			AddElement(/datum/element/dangerous_surgical_removal, fuse_time = 4 SECONDS, explosion_strength = 2, flag_blockers = ORGAN_FROZEN)
+			inflamation_stage = EXPLOSIVE_TERMINAL_STAGE + 1
+		if(4 to 5)
+			if(SPT_PROB(1.5, seconds_per_tick))
+				to_chat(organ_owner, span_warning("Your chest feels like it's barely able to keep something inside at bay, you feel extremely nauseous..."))
+				INVOKE_ASYNC(organ_owner, TYPE_PROC_REF(/mob, emote), "sway")
+				organ_owner.adjustOrganLoss(ORGAN_SLOT_APPENDIX, 5)
+				organ_owner.Stun(rand(40, 60))
+				organ_owner.adjustToxLoss(1, forced = TRUE)
+		if(6)
+			if(SPT_PROB(0.5, seconds_per_tick))
+				organ_owner.visible_message(span_userdanger("[organ_owner] clutches at their lower abdomen!"), span_userdanger("You clutch at your lower abdomen in blinding pain! It feels like something is about to burst from your body!"))
+			else if(SPT_PROB(1.5, seconds_per_tick))
+				INVOKE_ASYNC(organ_owner, TYPE_PROC_REF(/mob, emote), "grimace")
+				to_chat(organ_owner, span_warning("It's hard to think about anything but the crippling pain in your abdomen."))
+				organ_owner.damageoverlaytemp = 60
+				organ_owner.update_damage_hud()
+			else if(SPT_PROB(2.5, seconds_per_tick))
+				INVOKE_ASYNC(organ_owner, TYPE_PROC_REF(/mob, emote), "sway")
+				to_chat(organ_owner, span_warning("You're becoming delirious from the explosive pain in your chest."))
+				organ_owner.set_eye_blur_if_lower(6 SECONDS * seconds_per_tick)
+		if(7)
+			INVOKE_ASYNC(src, PROC_REF(kaboom))
+			inflamation_stage = EXPLOSIVE_MAX_STAGE // nothing happens in 8
+
+/obj/item/organ/internal/appendix/proc/kaboom()
+	stoplag(60 SECONDS)
+	if(get_stabilized_status())
+		inflamation_stage = EXPLOSIVE_MAX_STAGE - 1
+		return to_chat(owner, span_notice("Your chest feels slightly stabilized. It still hurts like hell though.."))
+	playsound(get_turf(src), 'sound/effects/fuse.ogg', 80)
+	owner.visible_message(span_userdanger("A strange sizzling sound emanates from [owner]..."))
+	stoplag(5 SECONDS)
+	explosion(owner, devastation_range = 0, heavy_impact_range = 1, light_impact_range = 2, flame_range = 3, flash_range = 4, explosion_cause = src)
```

Comment:
the location of the explosion should be the appendix itself, rather than the owner. Also check if it hasn't been deleted.